File: PoseKit/Sources/PoseKit/Joint.swift
/// The 19 body joints tracked by Apple Vision's VNDetectHumanBodyPoseRequest.
/// Raw values are the JSON keys used in bundled reference pose files.
public enum Joint: String, CaseIterable, Codable, Sendable, Hashable {
    case nose, leftEye, rightEye, leftEar, rightEar
    case neck, leftShoulder, rightShoulder
    case leftElbow, rightElbow, leftWrist, rightWrist
    case root, leftHip, rightHip
    case leftKnee, rightKnee, leftAnkle, rightAnkle
}

File: android/posekit/src/main/kotlin/com/oerol/posekit/Joint.kt
package com.oerol.posekit

/**
 * The 19 body joints tracked by the pose detector. Constant names are the JSON
 * keys used in bundled reference pose files (mirrors PoseKit's Swift `Joint`
 * raw values exactly — do not rename).
 */
@Suppress("EnumEntryName")
enum class Joint {
    nose, leftEye, rightEye, leftEar, rightEar,
    neck, leftShoulder, rightShoulder,
    leftElbow, rightElbow, leftWrist, rightWrist,
    root, leftHip, rightHip,
    leftKnee, rightKnee, leftAnkle, rightAnkle;

    companion object {
        fun fromKey(key: String): Joint? = entries.firstOrNull { it.name == key }
    }
}
